## Final Project Submission

Please fill out:
* Student name: Jeremiah Waiguru
* Student pace: full time
* Scheduled project review date/time: 
* Instructor name: Mwikali
* Blog post URL: N/A


# Overview

This project analyzes movies datasets from various movie rating sites (through Exploratory Data Analysis), to assist the head of Microsoft's new movie studio  in deciding what type of films to create under their new venture.

# Business understanding

Microsoft faces the challenge of competing against industry giants armed with historical industry insights.
To thrive in this competitive industry , microsoft needs to strategically determine the types of movies that are most likely to succeed in terms of profitability,ratings, and optimal release timing.

Business understanding in this scenario involves microsoft recognizing the opportunity to enter the film industry, specifically by establishing a new movie studio .This decision is driven by observation that many successful companies are creating original video content, and microsoft wants to leverage its resources and brand to tap into this market



# Business Objectives

1. Find out if venturing into the firm industry is a viable business idea for microsoft
2. Find out what kind of genres Microsoft needs to create to maximize on profitability
3. Find out what is the cost of producing a movie 

# Data Understanding

In this project we will be analyzing data  from these datasets: Box office Mojo, IMDB(Internet Movie Database) and the numbers'.
We'll now load the datasets below;

# Load libraries


In [42]:
# importing packages that we will use in this project
import csv
import pandas as pd
# setting pandas display to avoid scientific notation in the dataframes
pd.options.display.float_format = '{:.2f}'.format
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline
import sqlite3
import gzip
# ignore warnings
import warnings
warnings.filterwarnings('ignore')

# 1. Work on the bom.movie_gross.csv dataset

In [4]:
# loading the dataset
movie_gross = pd.read_csv('zippedData/bom.movie_gross.csv.gz')
movie_gross

title      studio  domestic_gross  \
0                                     Toy Story 3          BV    415000000.00   
1                      Alice in Wonderland (2010)          BV    334200000.00   
2     Harry Potter and the Deathly Hallows Part 1          WB    296000000.00   
3                                       Inception          WB    292600000.00   
4                             Shrek Forever After        P/DW    238700000.00   
...                                           ...         ...             ...   
3382                                    The Quake       Magn.         6200.00   
3383                  Edward II (2018 re-release)          FM         4800.00   
3384                                     El Pacto        Sony         2500.00   
3385                                     The Swan  Synergetic         2400.00   
3386                            An Actor Prepares       Grav.         1700.00   

     foreign_gross  year  
0        652000000  2010  
1        691300000  2010  
2        664300000  2010  
3        535700000  2010  
4        513900000  2010  
...            ...   ...  
3382           NaN  2018  
3383           NaN  2018  
3384           NaN  2018  
3385           NaN  2018  
3386           NaN  2018  

[3387 rows x 5 columns]

# Breakdown to understand our data

The DataFrame has 3387 rows and 5 columns.
## Columns:
1. title - the title of the movie
2. studio - where the movie was produced
3. domestic_gross - the domestic gross revenue of the movie in dollars
4. foreign_gross - the foreign gross revenue of the movie in dollars
5. year - the year in which the movie was produced

In [5]:
movie_gross.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


In [6]:
movie_gross.isna().any()

title             False
studio             True
domestic_gross     True
foreign_gross      True
year              False
dtype: bool

In [7]:
movie_gross.isna().sum()

title                0
studio               5
domestic_gross      28
foreign_gross     1350
year                 0
dtype: int64

In [27]:
movie_gross.duplicated()

0       False
1       False
2       False
3       False
4       False
        ...  
3382    False
3383    False
3384    False
3385    False
3386    False
Length: 3387, dtype: bool

### from the above information

1. we see the columns have different data types; object, float and integer
2. studio column, domestic_gross and foreign_gross  has missing data
3. foreign_gross has object as the data type yet it should be in float data type

N/B: missing data and the data type problem will be dealt with while cleaning the data

### summary statistics of the domestic gross

In [9]:
# descriptive statistics 
movie_gross['domestic_gross'].describe()

count        3359.00
mean     28745845.07
std      66982498.24
min           100.00
25%        120000.00
50%       1400000.00
75%      27900000.00
max     936700000.00
Name: domestic_gross, dtype: float64

1. The mean of the  columm is approximately 28.75 million dollars
2. The standard deviation is approximately 66.98 million dollars
3. Minimum value in the column is 1oo dollars, this shows that some movies in the dataset made very little money
4. Maximum value in the column is 936.7 million dollars to show that some movies in the dataset made huge profits domestically


In [10]:
movie_gross['foreign_gross'].describe()

count        2037
unique       1204
top       1200000
freq           23
Name: foreign_gross, dtype: object

1. The count is 2037 non_null values  meaning there are missing values  since we expect our count to be 3387.
2. The unique count shows that there are unique values in the column adding upto 1204
3. Top value in the column is 1.2 million dollars and it appears 23 times meaning there are 23 movies that made 1.2 million dollars in the foreign markets
4. freq shows the number of times a top value appears in column.

# 2. Work on the tmdb.movies dataset

In [8]:
# loading the dataset
tmdb_movie = pd.read_csv("C:\\Users\\JEREMY\\Downloads\\MORINGA 1\\dsc-phase-1-project-v2-4\\zippedData\\tmdb.movies.csv.gz", index_col=0)
tmdb_movie

genre_ids      id original_language  \
0          [12, 14, 10751]   12444                en   
1      [14, 12, 16, 10751]   10191                en   
2            [12, 28, 878]   10138                en   
3          [16, 35, 10751]     862                en   
4            [28, 878, 12]   27205                en   
...                    ...     ...               ...   
26512             [27, 18]  488143                en   
26513             [18, 53]  485975                en   
26514         [14, 28, 12]  381231                en   
26515      [10751, 12, 28]  366854                en   
26516             [53, 27]  309885                en   

                                     original_title  popularity release_date  \
0      Harry Potter and the Deathly Hallows: Part 1       33.53   2010-11-19   
1                          How to Train Your Dragon       28.73   2010-03-26   
2                                        Iron Man 2       28.52   2010-05-07   
3                                         Toy Story       28.00   1995-11-22   
4                                         Inception       27.92   2010-07-16   
...                                             ...         ...          ...   
26512                         Laboratory Conditions        0.60   2018-10-13   
26513                               _EXHIBIT_84xxx_        0.60   2018-05-01   
26514                                  The Last One        0.60   2018-10-01   
26515                                  Trailer Made        0.60   2018-06-22   
26516                                    The Church        0.60   2018-10-05   

                                              title  vote_average  vote_count  
0      Harry Potter and the Deathly Hallows: Part 1          7.70       10788  
1                          How to Train Your Dragon          7.70        7610  
2                                        Iron Man 2          6.80       12368  
3                                         Toy Story          7.90       10174  
4                                         Inception          8.30       22186  
...                                             ...           ...         ...  
26512                         Laboratory Conditions          0.00           1  
26513                               _EXHIBIT_84xxx_          0.00           1  
26514                                  The Last One          0.00           1  
26515                                  Trailer Made          0.00           1  
26516                                    The Church          0.00           1  

[26517 rows x 9 columns]

In [9]:
tmdb_movie.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 26517 entries, 0 to 26516
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   genre_ids          26517 non-null  object 
 1   id                 26517 non-null  int64  
 2   original_language  26517 non-null  object 
 3   original_title     26517 non-null  object 
 4   popularity         26517 non-null  float64
 5   release_date       26517 non-null  object 
 6   title              26517 non-null  object 
 7   vote_average       26517 non-null  float64
 8   vote_count         26517 non-null  int64  
dtypes: float64(2), int64(2), object(5)
memory usage: 2.0+ MB


In [10]:
tmdb_movie.isna().any()

genre_ids            False
id                   False
original_language    False
original_title       False
popularity           False
release_date         False
title                False
vote_average         False
vote_count           False
dtype: bool

In [11]:
tmdb_movie.isna().sum()

genre_ids            0
id                   0
original_language    0
original_title       0
popularity           0
release_date         0
title                0
vote_average         0
vote_count           0
dtype: int64

In [28]:
tmdb_movie.duplicated()

0        False
1        False
2        False
3        False
4        False
         ...  
26512    False
26513    False
26514    False
26515    False
26516    False
Length: 26517, dtype: bool

from the above information;
Our dataset is clean, no missing  data

# 3. work on movie budgets dataset

In [12]:
# load the data
movie_budgets = pd.read_csv("C:\\Users\\JEREMY\\Downloads\\MORINGA 1\\dsc-phase-1-project-v2-4\\zippedData\\tn.movie_budgets.csv.gz")
movie_budgets

id  release_date                                        movie  \
0      1  Dec 18, 2009                                       Avatar   
1      2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2      3   Jun 7, 2019                                 Dark Phoenix   
3      4   May 1, 2015                      Avengers: Age of Ultron   
4      5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   
...   ..           ...                                          ...   
5777  78  Dec 31, 2018                                       Red 11   
5778  79   Apr 2, 1999                                    Following   
5779  80  Jul 13, 2005                Return to the Land of Wonders   
5780  81  Sep 29, 2015                         A Plague So Pleasant   
5781  82   Aug 5, 2005                            My Date With Drew   

     production_budget domestic_gross worldwide_gross  
0         $425,000,000   $760,507,625  $2,776,345,279  
1         $410,600,000   $241,063,875  $1,045,663,875  
2         $350,000,000    $42,762,350    $149,762,350  
3         $330,600,000   $459,005,868  $1,403,013,963  
4         $317,000,000   $620,181,382  $1,316,721,747  
...                ...            ...             ...  
5777            $7,000             $0              $0  
5778            $6,000        $48,482        $240,495  
5779            $5,000         $1,338          $1,338  
5780            $1,400             $0              $0  
5781            $1,100       $181,041        $181,041  

[5782 rows x 6 columns]

In [13]:
movie_budgets.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


In [14]:
movie_budgets.isna().any()

id                   False
release_date         False
movie                False
production_budget    False
domestic_gross       False
worldwide_gross      False
dtype: bool

In [15]:
movie_budgets.isna().sum()

id                   0
release_date         0
movie                0
production_budget    0
domestic_gross       0
worldwide_gross      0
dtype: int64

In [29]:
movie_budgets.duplicated()

0       False
1       False
2       False
3       False
4       False
        ...  
5777    False
5778    False
5779    False
5780    False
5781    False
Length: 5782, dtype: bool

Likewise, the above dataset is clean with no missing data

## 4.  Work on rt reviews dataset

In [53]:
rt_movie = pd.read_csv('zippedData/rt.movie_info.tsv.gz', delimiter='\t', encoding='unicode_escape')
rt_movie.head()

id                                           synopsis rating  \
0   1  This gritty, fast-paced, and innovative police...      R   
1   3  New York City, not-too-distant-future: Eric Pa...      R   
2   5  Illeana Douglas delivers a superb performance ...      R   
3   6  Michael Douglas runs afoul of a treacherous su...      R   
4   7                                                NaN     NR   

                                 genre          director  \
0  Action and Adventure|Classics|Drama  William Friedkin   
1    Drama|Science Fiction and Fantasy  David Cronenberg   
2    Drama|Musical and Performing Arts    Allison Anders   
3           Drama|Mystery and Suspense    Barry Levinson   
4                        Drama|Romance    Rodney Bennett   

                            writer  theater_date      dvd_date currency  \
0                   Ernest Tidyman   Oct 9, 1971  Sep 25, 2001      NaN   
1     David Cronenberg|Don DeLillo  Aug 17, 2012   Jan 1, 2013        $   
2                   Allison Anders  Sep 13, 1996  Apr 18, 2000      NaN   
3  Paul Attanasio|Michael Crichton   Dec 9, 1994  Aug 27, 1997      NaN   
4                     Giles Cooper           NaN           NaN      NaN   

  box_office      runtime             studio  
0        NaN  104 minutes                NaN  
1    600,000  108 minutes  Entertainment One  
2        NaN  116 minutes                NaN  
3        NaN  128 minutes                NaN  
4        NaN  200 minutes                NaN

In [54]:
rt_movie.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1560 entries, 0 to 1559
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   id            1560 non-null   int64 
 1   synopsis      1498 non-null   object
 2   rating        1557 non-null   object
 3   genre         1552 non-null   object
 4   director      1361 non-null   object
 5   writer        1111 non-null   object
 6   theater_date  1201 non-null   object
 7   dvd_date      1201 non-null   object
 8   currency      340 non-null    object
 9   box_office    340 non-null    object
 10  runtime       1530 non-null   object
 11  studio        494 non-null    object
dtypes: int64(1), object(11)
memory usage: 146.4+ KB


In [55]:
rt_movie.isna().any()

id              False
synopsis         True
rating           True
genre            True
director         True
writer           True
theater_date     True
dvd_date         True
currency         True
box_office       True
runtime          True
studio           True
dtype: bool

In [57]:
rt_movie.isna().sum()

id                 0
synopsis          62
rating             3
genre              8
director         199
writer           449
theater_date     359
dvd_date         359
currency        1220
box_office      1220
runtime           30
studio          1066
dtype: int64

# Data Cleaning

Here we are going to clean the data to make it meaningfull for analysis

## Cleaning the Movie_gross dataset

In [22]:
# first we convert the data type from object to float
movie_gross['foreign_gross'] = pd.to_numeric(movie_gross['foreign_gross'],errors='coerce')
movie_gross

title      studio  domestic_gross  \
0                                     Toy Story 3          BV    415000000.00   
1                      Alice in Wonderland (2010)          BV    334200000.00   
2     Harry Potter and the Deathly Hallows Part 1          WB    296000000.00   
3                                       Inception          WB    292600000.00   
4                             Shrek Forever After        P/DW    238700000.00   
...                                           ...         ...             ...   
3382                                    The Quake       Magn.         6200.00   
3383                  Edward II (2018 re-release)          FM         4800.00   
3384                                     El Pacto        Sony         2500.00   
3385                                     The Swan  Synergetic         2400.00   
3386                            An Actor Prepares       Grav.         1700.00   

      foreign_gross  year  
0      652000000.00  2010  
1      691300000.00  2010  
2      664300000.00  2010  
3      535700000.00  2010  
4      513900000.00  2010  
...             ...   ...  
3382            nan  2018  
3383            nan  2018  
3384            nan  2018  
3385            nan  2018  
3386            nan  2018  

[3387 rows x 5 columns]

In [23]:
# confirm if now the foreign_gross column will give us descriptive statistics values
movie_gross['foreign_gross'].describe()

count        2032.00
mean     75057041.63
std     137529351.20
min           600.00
25%       3775000.00
50%      18900000.00
75%      75050000.00
max     960500000.00
Name: foreign_gross, dtype: float64

## lets now replace the missing data

In [24]:
# Replacing missing values in the studio column
movie_gross['studio'].fillna('None', inplace = True)

# Replacing missing values in domestic_gross column
movie_gross['domestic_gross'].fillna(0, inplace = True)

# Replacing missing values in foreign_gross column
movie_gross['foreign_gross'].fillna(0, inplace = True)

movie_gross


title      studio  domestic_gross  \
0                                     Toy Story 3          BV    415000000.00   
1                      Alice in Wonderland (2010)          BV    334200000.00   
2     Harry Potter and the Deathly Hallows Part 1          WB    296000000.00   
3                                       Inception          WB    292600000.00   
4                             Shrek Forever After        P/DW    238700000.00   
...                                           ...         ...             ...   
3382                                    The Quake       Magn.         6200.00   
3383                  Edward II (2018 re-release)          FM         4800.00   
3384                                     El Pacto        Sony         2500.00   
3385                                     The Swan  Synergetic         2400.00   
3386                            An Actor Prepares       Grav.         1700.00   

      foreign_gross  year  
0      652000000.00  2010  
1      691300000.00  2010  
2      664300000.00  2010  
3      535700000.00  2010  
4      513900000.00  2010  
...             ...   ...  
3382           0.00  2018  
3383           0.00  2018  
3384           0.00  2018  
3385           0.00  2018  
3386           0.00  2018  

[3387 rows x 5 columns]

In [25]:
# confirming there is no any missing data
movie_gross.isna().sum()

title             0
studio            0
domestic_gross    0
foreign_gross     0
year              0
dtype: int64

In [26]:
movie_gross.duplicated()

0       False
1       False
2       False
3       False
4       False
        ...  
3382    False
3383    False
3384    False
3385    False
3386    False
Length: 3387, dtype: bool

## lets clean movie budgets

In [31]:
movie_budgets.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


In [30]:
movie_budgets.isna().sum()

id                   0
release_date         0
movie                0
production_budget    0
domestic_gross       0
worldwide_gross      0
dtype: int64

from the information above, our dataset is clean but we now want to convert data type of production_budget, domestic_gross, worldwide_gross from object to float

In [37]:
# change data type of production budget to float
movie_budgets['production_budget'] = movie_budgets['production_budget'].replace('[^.0-9]', '', regex=True).astype(float)

# change data type of domestic_gross
movie_budgets['domestic_gross'] = movie_budgets['domestic_gross'].replace('[^.0-9]', '', regex=True).astype(float)

# change data type of worldwide_gross
movie_budgets['worldwide_gross'] = movie_budgets['worldwide_gross'].replace('[^.0-9]', '', regex=True).astype(float)

movie_budgets

id  release_date                                        movie  \
0      1  Dec 18, 2009                                       Avatar   
1      2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2      3   Jun 7, 2019                                 Dark Phoenix   
3      4   May 1, 2015                      Avengers: Age of Ultron   
4      5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   
...   ..           ...                                          ...   
5777  78  Dec 31, 2018                                       Red 11   
5778  79   Apr 2, 1999                                    Following   
5779  80  Jul 13, 2005                Return to the Land of Wonders   
5780  81  Sep 29, 2015                         A Plague So Pleasant   
5781  82   Aug 5, 2005                            My Date With Drew   

      production_budget  domestic_gross  worldwide_gross  
0          425000000.00    760507625.00    2776345279.00  
1          410600000.00    241063875.00    1045663875.00  
2          350000000.00     42762350.00     149762350.00  
3          330600000.00    459005868.00    1403013963.00  
4          317000000.00    620181382.00    1316721747.00  
...                 ...             ...              ...  
5777            7000.00            0.00             0.00  
5778            6000.00        48482.00        240495.00  
5779            5000.00         1338.00          1338.00  
5780            1400.00            0.00             0.00  
5781            1100.00       181041.00        181041.00  

[5782 rows x 6 columns]

In [38]:
movie_budgets.duplicated()

0       False
1       False
2       False
3       False
4       False
        ...  
5777    False
5778    False
5779    False
5780    False
5781    False
Length: 5782, dtype: bool

In [58]:
# drop all the duplicates
movie_budgets.drop_duplicates(subset='movie', keep='first', inplace=True)
movie_budgets

id  release_date                                        movie  \
0      1  Dec 18, 2009                                       Avatar   
1      2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2      3   Jun 7, 2019                                 Dark Phoenix   
3      4   May 1, 2015                      Avengers: Age of Ultron   
4      5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   
...   ..           ...                                          ...   
5777  78  Dec 31, 2018                                       Red 11   
5778  79   Apr 2, 1999                                    Following   
5779  80  Jul 13, 2005                Return to the Land of Wonders   
5780  81  Sep 29, 2015                         A Plague So Pleasant   
5781  82   Aug 5, 2005                            My Date With Drew   

      production_budget  domestic_gross  worldwide_gross  
0          425000000.00    760507625.00    2776345279.00  
1          410600000.00    241063875.00    1045663875.00  
2          350000000.00     42762350.00     149762350.00  
3          330600000.00    459005868.00    1403013963.00  
4          317000000.00    620181382.00    1316721747.00  
...                 ...             ...              ...  
5777            7000.00            0.00             0.00  
5778            6000.00        48482.00        240495.00  
5779            5000.00         1338.00          1338.00  
5780            1400.00            0.00             0.00  
5781            1100.00       181041.00        181041.00  

[5698 rows x 6 columns]